packages/desktop-client/src/components/FinancesApp.js

```diff
@@ -253,96 +299,87 @@ class FinancesApp extends React.Component {
     // Wait a little bit to make sure the sync button will get the
     // sync start event. This can be improved later.
     setTimeout(async () => {
-      await this.props.sync();
+      await props.sync();
 
       // Check for upgrade notifications. We do this after syncing
       // because these states are synced across devices, so they will
       // only see it once for this file
       checkForUpgradeNotifications(
-        this.props.addNotification,
-        this.props.resetSync,
-        this.history,
-      );
-    }, 100);
-
-    setTimeout(async () => {
-      await this.props.sync();
-      await checkForUpdateNotification(
-        this.props.addNotification,
-        getIsOutdated,
-        getLatestVersion,
-        this.props.loadPrefs,
-        this.props.savePrefs,
+        props.addNotification,
+        props.resetSync,
+        patchedHistory,
       );
     }, 100);
+  }, []);
+
+  setTimeout(async () => {
```

Comment:
This will happen on every render, it should be moved into a `useEffect` or something IMO

Reply:
Good call.

Reply:
Oh shoot - this duplicates the setTimeout I wrapped in a `useEffect` just above. De-duping.

Reply:
Moving it into the setTimeout inside the useEffect I added for all the rest of the side effects here. (Bad merge conflict resolution, I suspect--was duplicating `props.sync` calls).